examples/feature_demo/video_yuv.py

```diff
@@ -0,0 +1,176 @@
+"""
+Video yuv
+=========
+
+This example demonstrates how to show video data in a very efficient way,
+eliminating as much data-copies as we can.
+
+Reading
+-------
+
+Many videos are encoded using yuv420p. Therefore, loading the video in
+that format, instead of requesting rgb frames, safes computation and
+data copies upon reading.
+
+Uploading
+---------
+
+The YUV data can also be uploaded directly to the texture, albeit in
+three chunks. RGBA frames can be send in one chunk (although it is
+considerably larger). For RGB data, the data must first be converted
+to RGBA, which is costly. (This packing is normally done automatically
+by the pygfx Texture).
+
+Size
+----
+
+Frames in yuv420p format are half the size of RGB frames (and even less
+compared to RGBA) so the upload itself is cheaper, and memory
+consumption on the CPU and GPU are reduced.
+
+Benchmarks
+----------
+
+This script can be done to benchmark for different formats. We've found
+that using yuv420p data is certainly the fastest. Using rgba is a bit
```

Comment:
Ultimately the most important aspect is that the data storage format be aligned with the data display format. 

For example, if the video is encoded as RGBA (very rare in many consumer formats) it may be an advantage to operate the shader with RGBA. However, most video is encoded as YUV420 with limited color range. As such we found that ….

Reply:
Yeah, I understand. Though I was surprised by the relatively good performance of converting to rgba using av. It looks like the rgb-to-rgba conversation we do in pygfx is a tighter bottleneck.

Reply:
One of the big big challenges in all these libraries is assessing the single core performance (which for me is all I care about since at some point my problem is embarrassingly parallelizable) so it is possible to ffmpeg is using multiple threads to do its work. (This is Something I have tuned in the past) 

I mostly left this note here so I can help expand on this motivation a little bit when I get to a computer.

Reply:
Ah, so if I understand what you're saying correctly, the relatively good performance of letting ffmpeg convert to rgba for us, is likely because ffmpeg uses multithreading, and that "trick" loses its benefit when you want to load multiple frames in parallel.

Reply:
Multiple frames from multiple different video streams. Correct.